inst/include/common/fims_vector.hpp

```diff
@@ -0,0 +1,483 @@
+    void reserve(size_type cap) {
+        this->vec_m.reserve(cap);
+    }
+    
+    
+    /**
+     * @brief Returns the number of elements that can be held in currently allocated storage.
+     */
+    size_type capacity() {
+        return this->vec_m.capacity();
+    }
+    
+    /**
+     *  @brief Reduces memory usage by freeing unused memory.
+     */
+    void shrink_to_fit() {
+        this->vec_m.shrink_to_fit();
+    }
+    
+    //modifiers
+    
+    /**
+     * @brief Clears the contents.
+     */
+    void clear() {
+        this->vec_m.clear();
+    }
+    
+    
+    /**
+     * @brief Inserts value before pos.
+     */
+    iterator insert(const_iterator pos, const Type& value) {
+        return this->vec_m.insert(pos, value);
+    }
+    
+    /**
+     * @brief Inserts count copies of the value before pos.
+     */
+    iterator insert(const_iterator pos, size_type count, const Type& value) {
+        return this->vec_m.insert(pos, count, value);
+    }
+    
+    /**
+     * @brief Inserts elements from range [first, last) before pos.
+     */
+    template< class InputIt >
+    iterator insert(const_iterator pos, InputIt first, InputIt last) {
+        return this->vec_m.insert(pos, first, last);
+    }
+    
+    /**
+     * @brief Inserts elements from initializer list ilist before pos.
+     */
+    
+    iterator insert(const_iterator pos, std::initializer_list<Type> ilist) {
+        return this->vec_m.insert(pos, ilist);
+    }
+    
+    
+    /**
+     * @brief Constructs element in-place.
+     */
+    template< class... Args >
+    iterator emplace(const_iterator pos, Args&&... args) {
+        return this->vec_m.emplace(pos, std::forward<Args>(args)...);
+    }
+    
+    
+    /**
+     * @brief Removes the element at pos.
+     */
+    iterator erase(iterator pos) {
+        return this->vec_m.erase(pos);
+    }
+    
+    /**
+     * @brief Removes the elements in the range [first, last).
+     */
+    iterator erase(iterator first, iterator last) {
+        return this->vec_m.erase(first, last);
+    }
+    
+    
+    /**
+     * @brief Adds an element to the end.
+     */
+    void push_back(const Type&& value) {
+        this->vec_m.push_back(value);
+    }
+    
+    /**
+     * @brief Constructs an element in-place at the end.
+     */
+    template< class... Args >
+    void emplace_back(Args&&... args) {
+        this->vec_m.emplace_back(std::forward<Args>(args)...);
+    }
+    
+    /**
+     * @brief Removes the last element.
+     */
+    void pop_back() {
+        this->vec_m.pop_back();
+    }
+    
+    
+    /**
+     * @brief Changes the number of elements stored.
+     */
+    void resize(size_t s) {
+        this->vec_m.resize(s);
+    }
+    
+    /**
+     * @brief Swaps the contents.
+     */
+    void swap( Vector& other ){
+        this->vec_m.swap(other.vec_m);
+    }
+    
+    
+    //conversion operatrors
+    
+    /**
+     * @brief Converts this vector a std::vector<Type>
```

Comment:
change descriptions of conversion operators from `this vector` to `fims::Vector<Type>`